cpp/src/arrow/util/bit_util.h

```diff
@@ -1170,5 +1170,23 @@ class BitsetStack {
   std::vector<int> offsets_;
 };
 
+/// \brief A class that scans through a true/false bitmap to yield runs of
+/// all-true or not-all-true runs. This is used to accelerate processing of
+/// mostly-not-null array data. 4 words are examined at a time.
+struct ARROW_EXPORT BitmapScanner {
```

Comment:
why not e class?

Also as Antoine pointed out on the JIRA this replicates a lot of functionality in BitRunReader it would be nice consolidate the two (although for this specific use-case I can imagine that this would be slightly more performant).  I think all that would be missing from the current implementation in https://github.com/apache/arrow/pull/7143

Reply:
I'll change it to a class. This code is simple and extremely fast -- for this specific use case I don't want to use a more general solution if it reduces performance

Reply:
Turned it into a class.

Reply:
Sounds OK to me.  i think a better name for this would be something along the lines of BitBlockCounter or something similar.  BitmapScanner is too generic.

Reply:
Changing

Reply:
`BitBlockCounter` it is